src/data/model/LayerTreeNode.js

```diff
@@ -90,9 +98,67 @@ Ext.define('GeoExt.data.model.LayerTreeNode', {
         this.callParent(arguments);
 
         // forward changes to ol object
-        if (key === 'checked') {
+        if (!this.__bubbling && key === 'checked') {
+            if(this.get('isLayerGroup') && this.get('toggleMode') === 'classic'){
+                this.__updating = true;
+                this.getOlLayer().set('visible', newValue);
+                if(this.childNodes){
+                    this.eachChild(function(child){
+                        child.getOlLayer().set('visible', newValue);
+                    });
+                }
+                this.__updating = false;
+            } else {
+                this.__updating = true;
+                this.getOlLayer().set('visible', newValue);
+                this.__updating = false;
+            }
```

Comment:
I think the whole `if-else`-block can be wrapped with `this.__updating = true/false`, right? Would possibly enhance legibility.